src/modules/site-settings/edit-site-details.tsx

```diff
@@ -155,7 +156,7 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 				phpVersion: selectedPhpVersion,
 				isWpAutoUpdating: selectedWpVersion === DEFAULT_WORDPRESS_VERSION,
 				customDomain: usedCustomDomain,
-				enableHttps: !! usedCustomDomain && enableHttps,
+				enableHttps: !! usedCustomDomain && ( isCertificateTrusted || enableHttps ),
```

Comment:
```suggestion
				enableHttps: !! usedCustomDomain && enableHttps,
```
Why would we not respect the state of the `Enable HTTPS` checkbox when saving the site?

Reply:
I totally agree about that change. We should always honour the saved state.